File: jsweet-client/src/main/java/json_schema/events/ScannedBotEvent.java
package json_schema.events;

import static def.jquery.Globals.$;

import json_schema.types.Point;

public final class ScannedBotEvent extends Event {

	public static final String TYPE = "scanned-bot-event";

	public ScannedBotEvent() {
		super(TYPE);
	}

	public Integer getScannedByBotId() {
		return (Integer) $get("scanned-by-bot-id");
	}

	public Integer getScannedBotId() {
		return (Integer) $get("scanned-bot-id");
	}

	public Double getEnergy() {
		return (Double) $get("energy");
	}

	public Point getPosition() {
		return (Point) $.extend(false, new Point(), $get("position"));
	}

	public Double getDirection() {
		return (Double) $get("direction");
	}

	public Double getSpeed() {
		return (Double) $get("speed");
	}
}

File: bot-api/python/tankroyale/botapi/events/ScannedBotEvent.py
import BotEvent


class ScannedBotEvent(BotEvent):
    scanned_by_bot_id: int
    scanned_bot_id: int
    energy: float
    x: float
    y: float
    direction: float
    speed: float

    def __init__(self,
                 turn_number: int,
                 scanned_by_bot_id: int,
                 scanned_bot_id: int,
                 energy: float,
                 x: float,
                 y: float,
                 direction: float,
                 speed: float):
        super(turn_number)
        self.scanned_by_bot_id = scanned_by_bot_id
        self.scanned_bot_id = scanned_bot_id
        self.energy = energy
        self.x = x
        self.y = y
        self.direction = direction
        self.speed = speed

    def scanned_by_bot_id(self):
        return self.scanned_by_bot_id

    def scanned_bot_id(self):
        return self.scanned_bot_id

    def energy(self):
        return self.energy

    def x(self):
        return self.x

    def y(self):
        return self.y

    def direction(self):
        return self.direction

    def speed(self):
        return self.speed
